### ISSUE TITLE
Suggestion: Story settings the second

### ISSUE THREAD

**Issue by Muskelmagier** [cloveranon/Clover-Edition#108]
I am sorry that I bring this up a second time, but Clover editions has too many settings for them to be not better explained.

the OpenCYOAI wiki is not enough, because Clover edition has custom settings where if they are set on default it only produces nonsense.

action-temp        How random the suggested actions are. Default: 1 
produces replies  which are more like a deranged preteen rapper; if you use the default of 1

I tried to do a Touhou style story and this is the first reply with action-temp default (1)

_You appear in Gensokyo, in the distance you see the small hakurei shrine. 
Inside, several lacquer work loli kobold in flower costumes do useless mental 
exercises to herself. Suddenly, two blurpy swords impaled into your own arms... 
◙Three Wankers Reborn into Magic (Six Demons+Shintoi Hatuke!)You struggle to 
avoid most monsters because of its claws or snake like eyes._

what could be done to fix this:

1. How about a few preset suggestions for different story type, like bundling settings together for a preset for more focused stories like a focused scene or another preset for adventure stories which can be more random. 
2. More sensible default settings like I said above.

I am only bringing this it is something that can be done better.

**Comment by Akababa** [comment]
Sounds reasonable. Actually the "defaults" shown in the game aren't defaults at all, since the real defaults are in config.ini. We should probably remove that file from the repo and generate it on first run. BTW config.ini actually uses temp 0.21, not 1.
You can update this file with your settings and submit a PR:
https://github.com/cloveranon/Clover-Edition/blob/master/getconfig.py#L39

**Comment by MikkoMMM** [comment]
I think we should refactor the instructions system internally at the very least. Various instructions are in several places right now (not as maintainable as they could be). Also, we should have some kind of a default configuration file which is used for creating a user-specific configuration files and never modified. Maybe the "default values" in the instructions could be read from that, too.

**Comment by MikkoMMM** [comment]
I did a bit of reworking to config.ini already in https://github.com/cloveranon/Clover-Edition/commit/d6614f4dbb87ae805a8abc12864439e4e5a567e0#diff-38589f3a5b1e8e2eca835f956b204e0e
But I probably don't understand the settings any better than you do. It's a reasonable request make the explanations clearer in my opinion as well. That also brought into my mind that our current code is very undocumented internally as well which I'm planning to do some little improvements in... at some point. Hopefully.

**Comment by Muskelmagier** [comment]
I would propose then we find better default values for everything.
I few settings are already determined by other users, but things like top-keks, action-temp, log-level, and rep-pen would good to find appropriate default values.

**Comment by cloveranon** [comment]
So here's what I'm working on, I make no promises I'll finish it any time soon. But you can see in the experimental branch I've got it to generate multiple replies to every input. And the user selects the best reply. There was something like this in AI Dungeon 1.

Now what I want to do is randomize the settings every time it generates a reply. E.g. the first choice will use a temp of 0.2, the second choice will use a temp of 0.3. These will not be shown to the user, so it will be a fair blind test. It will record your choice, and we can get statistics on what the best settings are.

I plan to introduce a ton more settings for super fine grained control of the AI's output. E.g. I have half a dozen word penalties on the experimental branch that do things like decrease the average length of it's responses, make it less likely to produce dialogue, make it less likely to end the game, make it less likely to describe the player's actions directly, etc. Guessing what the number for all these different settings should be is hard and easy to screw up.

**Comment by MikkoMMM** [comment]
I kind of liked how the Clover Edition doesn't even have any of that cloud storage thing, so there's some peace of mind that whatever I do locally stays local. I'm not against the sort of features that require sending of data, but it shouldn't be on by default. Though, we can ask for permission to enable the gathering of anonymized statistics, which seems to be what you might be suggesting. And even then I'm no information security expert so I at least am not qualified for reviewing such features.

**Comment by Akababa** [comment]
I agree it should be opt-in, also it's not as big of a deal sending choices vs. sending the entire story.

**Comment by Muskelmagier** [comment]
Yeah thinking about the insane amount of Porn you would have to read through it really should only be opt-in

**Comment by VBPXKSMI** [comment]
Adding my opinion, I too think it should be opt-in.
Not relying on third parties is (one) of the greatest strenghts of this Anon's project over the mormon's ones, so privacy is kind of a "big" issue. But yeah, the feature and its possibilities should be explored as much as possible as long as it's completly optional (and tampering proof I should add).

**Comment by Skhmt** [comment]
> I kind of liked how the Clover Edition doesn't even have any of that cloud storage thing, so there's some peace of mind that whatever I do locally stays local. 

Just wanted to note how prescient this was.

**Comment by MikkoMMM** [comment]
Those who would give up essential safety, to purchase a little temporary convenience, deserve neither safety nor convenience.

Now hopefully this surge of renewed activity is here to stay for a while. A bit silly how it so often requires a major scandal for people to wake up.
